homeassistant/components/sensor/min_max.py

```diff
@@ -98,7 +100,7 @@ def async_min_max_sensor_state_listener(entity, old_state, new_state):
             if self._unit_of_measurement != new_state.attributes.get(
                     ATTR_UNIT_OF_MEASUREMENT):
                 _LOGGER.warning("Units of measurement do not match")
-                return
```

Comment:
> either the one configured with this sensor

The `min_max` sensor has no option to set the `unit_of_measurement` by itself.

> first incoming unit is wrong and remembered by this sensor. when it's fixed in config and config is reloaded, it won't work

If you get a unit of measurement mismatch and fix that then the sensor will work. The `min_max` sensor like all other sensors don't know what was stored before the HA restart.

> ignore units, use only the unig configured for min_max

As I told before, not a smart way to go as the result will become unpredictable. If you want to do that you can use a `template` sensor first and feed that to the `min_max` sensor.

> remember units for all values. if all matches, use this unit. otherwise return ERROR as unit, that's clearly visible in the UI

I decided to use the first entity as lead for the unit of measurement. Sure if the first one is not what you want the sensor assume that the rest of the entities is wrong. Humans can deal with false-positives pretty quick.

Well, the log file is for errors and warning. We could create a persistent notification and add the `entity_id` to the warning. Add the `entity_id` would be an improvement anyway.